harness/determined/common/experimental/trial.py

```diff
@@ -185,13 +189,32 @@ def logs(
                 specific ranks.  Defaults to ``None``.
             stdtypes (List[int], optional): When set, only fetch logs from lines from the given
                 stdio outputs.  Defaults to ``None`` (same as ``["stdout", "stderr"]``).
-            min_level: (LogLevel, optional): When set, defines the minimum log priority for lines
+            min_level (LogLevel, optional): When set, defines the minimum log priority for lines
                 that will be returned.  Defaults to ``None`` (all logs returned).
+            timestamp_before (Union[str, int], optional): Specifies a timestamp that filters logs
+                before a certain time. Accepts either a string in RFC 3339 format
+                (eg. ``2021-10-26T23:17:12Z``) or an int representing the epoch second.
+            timestamp_after (Union[str, int], optional): Specifies a timestamp that filters logs
+                after a certain time. Accepts either a string in RFC 3339 format
+                (eg. ``2021-10-26T23:17:12Z``) or an int representing the epoch second.
+            sources (List[str], optional): Specifies a filter for logs based on originating node
+                name (eg. ``master`` or ``agent``).
+            search_text (str, Optional): Filters individual logs by searching for a string.
+
         """
         if head is not None and head < 0:
             raise ValueError(f"head must be non-negative, got {head}")
         if tail is not None and tail < 0:
             raise ValueError(f"tail must be non-negative, got {tail}")
+
+        # Convert epoch timestamps to RFC 3339-formatted datetime strings.
+        if isinstance(timestamp_before, int):
```

Comment:
Some of these parameters are tricky (not just `timestamp_*`). And while it would have been ideal to have validation for the rest of them, I think it's worth adding for `timestamp_*` even if they're the only ones.

Reply:
that's a very good point. added an additional validation for accepted types.